internal/exec/stages/disks/disks.go

```diff
@@ -257,32 +257,45 @@ func (s stage) createFilesystems(config types.Config) error {
 }
 
 func (s stage) createFilesystem(fs types.Mount) error {
-	if fs.Create == nil {
+	format, err := s.readFilesystemType(fs)
+	if err != nil {
+		return err
+	}
+	if format == fs.Format && !fs.WipeFilesystem {
+		s.Logger.Info("filesystem at %q is already formatted. Skipping mkfs...", fs.Device)
 		return nil
 	}
 
 	mkfs := ""
-	args := translateV2_1OptionSliceToStringSlice(fs.Create.Options)
+	var force bool
+	var args []string
+	if fs.Create == nil {
+		args = translateMountOptionSliceToStringSlice(fs.Options)
+	} else {
+		force = fs.Create.Force
+		args = translateCreateOptionSliceToStringSlice(fs.Create.Options)
+	}
+	force = force || fs.WipeFilesystem // we must force if we're wiping the filesystem
```

Comment:
Since force is gonna be deprecated, the function should be called `wipe` and not force.